### ISSUE TITLE
I/O Error [Unable to retrieve container name for [<container>] : Premature end of chunk coded message body: closing chunk expected

### ISSUE THREAD

**Issue by emetsger** [fabric8io/docker-maven-plugin#1453]
### Description
`docker:start` fails with:
```
[ERROR] DOCKER> I/O Error [Unable to retrieve container name for [elasticsearch] : Premature end of chunk coded message body: closing chunk expected]
```

on MacOS:
* docker-maven-plugin version: 0.35.0 (also tried 0.30.0, which fails as well)
* Docker version: engine 19.03.13, desktop (for mac) 2.5.0.1
```
Client: Docker Engine - Community
 Cloud integration: 1.0.2
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
* Maven version (`mvn -v`) :
```
Apache Maven 3.5.0 (ff8f5e7444045639af65f6095c62210b5713f426; 2017-04-03T15:39:06-04:00)
Maven home: /Users/esm/maven-3
Java version: 1.8.0_201, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_201.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.14.6", arch: "x86_64", family: "mac"
```

To reproduce: 
* clone https://github.com/OA-PASS/nihms-submission-etl
* run `mvn clean install`

Note:
* The examples and output below come from executing `mvn` within the `nihms-etl-integration` sub-project.  
> Executing `mvn clean install` from the top level project is only necessary the first time you clone the project; after which the artifacts are in your local repository.
* The example and output below come from executing `mvn` on a [PR](https://github.com/OA-PASS/nihms-submission-etl/pull/49) that was submitted against `master`, but I can confirm the issue shows up against `master` (the PR does not introduce the issue being reported here).

### Summary
* Travis builds succeed, and builds on a colleague's machine succeed, but local builds on my platform do not.
  * ✅ Travis uses version 0.30.0 of the DMP, Java 8, and docker version 17.09.0-ce (api version 1.32)
  * ✅ Colleague's build uses version 0.30.0 of the DMP, Java 11, and docker version 20.10.5 (api version 1.41)
  * ❌ My local build uses 0.30.0 of the DMP (but I also tried 0.35.0 as well), Java 8 and docker version 19.03.13 (api version 1.40)

### Remediation Attempts
Here are things I've attempted to do with my local build to get things to work:

* Upgrading to version `0.35.0` of the plugin didn't help
* Use a different Docker API version
  * ⚠️ Using `<apiVersion>` or `docker.apiVersion` to downgrade the API to `1.32` (the version used by Travis) did not seem to work.  Despite specifying the API parameter, the output of  `-Ddocker.verbose=all` indicated that version `1.40` of the API was still being used
    * This was true for both version `0.35.0` and `0.30.0` of the plugin; I can open up a separate issue for this if necessary.
* Used `curl` to connect to `/var/run/docker.sock` and emulate the behavior of the DMP, but did not see anything weird in the responses

#### ❌ Fails locally on my platform
(see full [build log](https://gist.github.com/emetsger/bb607fbefedac9b43d628c131b9bdf4d))

* Java 8
* docker-maven-plugin version: 0.35.0 (also tried 0.30.0, which fails as well)
* Docker version: engine 19.03.13, desktop (for mac) 2.5.0.1
```
Client: Docker Engine - Community
 Cloud integration: 1.0.2
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
* Maven version (`mvn -v`) :
```
Apache Maven 3.5.0 (ff8f5e7444045639af65f6095c62210b5713f426; 2017-04-03T15:39:06-04:00)
Maven home: /Users/esm/maven-3
Java version: 1.8.0_201, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_201.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.14.6", arch: "x86_64", family: "mac"
```

Maven output (elided for brevity) of `mvn -e -Ddocker.verbose=all verify`:
```
[INFO] --- docker-maven-plugin:0.30.0:start (start-docker-its) @ nihms-etl-integration ---
[INFO] DOCKER> POSTing to unix://127.0.0.1:1/v1.40/networks/create with {"Name":"pass"}
[INFO] DOCKER> POSTing to unix://127.0.0.1:1/v1.40/containers/create?name=fcrepo with {"Image":"oapass/fcrepo:4.7.5-3.2-5","Hostname":"fcrepo","ExposedPorts":{"54986/tcp":{}},"Env":["FCREPO_JMS_PORT=54988","COMPACTION_URI=https://oa-pass.github.io/pass-data-model/src/main/resources/context-3.1.jsonld","COMPACTION_PRELOAD_FILE_PASS_STATIC=/usr/local/tomcat/lib/context-3.1.jsonld","COMPACTION_PRELOAD_URI_PASS_STATIC=https://oa-pass.github.io/pass-data-model/src/main/resources/context-3.1.jsonld","FCREPO_HOST=fcrepo","FCREPO_ACTIVEMQ_CONFIGURATION=classpath:/activemq-queue.xml","FCREPO_PORT=54986","FCREPO_JMS_BASEURL=http://localhost:54986/fcrepo/rest","FCREPO_STOMP_PORT=54989"],"Labels":{"dmp.coordinates":"org.dataconservancy.pass:nihms-etl-integration:1.4.1-SNAPSHOT"},"HostConfig":{"PortBindings":{"54986/tcp":[{"HostPort":"54986"}]},"Binds":["/Users/esm/workspaces/pass/nihms-submission-etl/nihms-etl-integration/src/test/resources/docker/mnt:/mnt"],"VolumesFrom":[],"NetworkMode":"pass"},"Volumes":{"/mnt":{}},"NetworkingConfig":{"EndpointsConfig":{"pass":{"Aliases":["fcrepo","localhost"]}}}}
[INFO] DOCKER> [oapass/fcrepo:4.7.5-3.2-5] "fcrepo": Start container 715c1486038b
[INFO] DOCKER> [oapass/fcrepo:4.7.5-3.2-5] "fcrepo": Waiting on url http://fedoraAdmin:moo@localhost:54986/fcrepo/rest with method HEAD for status 200..399.
[INFO] DOCKER> [oapass/fcrepo:4.7.5-3.2-5] "fcrepo": Waited on url http://fedoraAdmin:moo@localhost:54986/fcrepo/rest 33018 ms
[ERROR] DOCKER> Error occurred during container startup, shutting down...
[INFO] DOCKER> DELETEing unix://127.0.0.1:1/v1.40/containers/715c1486038b?v=0
[INFO] DOCKER> [oapass/fcrepo:4.7.5-3.2-5] "fcrepo": Stop and removed container 715c1486038b after 0 ms
[INFO] DOCKER> DELETEing unix://127.0.0.1:1/v1.40/networks/3aabb87a5f37
[ERROR] DOCKER> I/O Error [Unable to retrieve container name for [elasticsearch] : Premature end of chunk coded message body: closing chunk expected]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 47.216 s
[INFO] Finished at: 2021-04-15T10:47:06-04:00
[INFO] Final Memory: 45M/633M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.30.0:start (start-docker-its) on project nihms-etl-integration: I/O Error: Unable to retrieve container name for [elasticsearch] : Premature end of chunk coded message body: closing chunk expected -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:docker-maven-plugin:0.30.0:start (start-docker-its) on project nihms-etl-integration: I/O Error
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: I/O Error
	at io.fabric8.maven.docker.StartMojo.executeInternal(StartMojo.java:194)
	at io.fabric8.maven.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:230)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
Caused by: io.fabric8.maven.docker.access.DockerAccessException: Unable to retrieve container name for [elasticsearch] : Premature end of chunk coded message body: closing chunk expected
	at io.fabric8.maven.docker.access.hc.DockerAccessWithHcClient.inspectContainer(DockerAccessWithHcClient.java:355)
	at io.fabric8.maven.docker.access.hc.DockerAccessWithHcClient.getContainer(DockerAccessWithHcClient.java:332)
	at io.fabric8.maven.docker.service.QueryService.getContainer(QueryService.java:51)
	at io.fabric8.maven.docker.service.RunService.findContainerId(RunService.java:428)
	at io.fabric8.maven.docker.service.RunService.findContainerIdsForLinks(RunService.java:393)
	at io.fabric8.maven.docker.service.RunService.createContainerHostConfig(RunService.java:338)
	at io.fabric8.maven.docker.service.RunService.createContainerConfig(RunService.java:301)
	at io.fabric8.maven.docker.service.RunService.createAndStartContainer(RunService.java:150)
	at io.fabric8.maven.docker.service.helper.StartContainerExecutor.startContainers(StartContainerExecutor.java:47)
	at io.fabric8.maven.docker.StartMojo.lambda$startImage$0(StartMojo.java:304)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:117)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:38)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:77)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:260)
	at java.util.concurrent.ExecutorCompletionService.submit(ExecutorCompletionService.java:181)
	at io.fabric8.maven.docker.StartMojo.startImage(StartMojo.java:298)
	at io.fabric8.maven.docker.StartMojo.executeInternal(StartMojo.java:166)
	... 23 more
Caused by: org.apache.http.ConnectionClosedException: Premature end of chunk coded message body: closing chunk expected
	at org.apache.http.impl.io.ChunkedInputStream.getChunkSize(ChunkedInputStream.java:266)
	at org.apache.http.impl.io.ChunkedInputStream.nextChunk(ChunkedInputStream.java:225)
	at org.apache.http.impl.io.ChunkedInputStream.read(ChunkedInputStream.java:184)
	at org.apache.http.conn.EofSensorInputStream.read(EofSensorInputStream.java:135)
	at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.io.InputStreamReader.read(InputStreamReader.java:184)
	at java.io.Reader.read(Reader.java:140)
	at org.apache.http.util.EntityUtils.toString(EntityUtils.java:227)
	at org.apache.http.util.EntityUtils.toString(EntityUtils.java:308)
	at io.fabric8.maven.docker.access.hc.ApacheHttpClientDelegate.getResponseMessage(ApacheHttpClientDelegate.java:87)
	at io.fabric8.maven.docker.access.hc.ApacheHttpClientDelegate.access$000(ApacheHttpClientDelegate.java:20)
	at io.fabric8.maven.docker.access.hc.ApacheHttpClientDelegate$BodyAndStatusResponseHandler.handleResponse(ApacheHttpClientDelegate.java:189)
	at io.fabric8.maven.docker.access.hc.ApacheHttpClientDelegate$BodyAndStatusResponseHandler.handleResponse(ApacheHttpClientDelegate.java:183)
	at io.fabric8.maven.docker.access.hc.ApacheHttpClientDelegate$StatusCodeCheckerResponseHandler.handleResponse(ApacheHttpClientDelegate.java:172)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:223)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:165)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:140)
	at io.fabric8.maven.docker.access.hc.ApacheHttpClientDelegate.get(ApacheHttpClientDelegate.java:74)
	at io.fabric8.maven.docker.access.hc.DockerAccessWithHcClient.inspectContainer(DockerAccessWithHcClient.java:353)
	... 41 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
esm:~/workspaces/pass/nihms-submission-etl/nihms-etl-integration (detached*)*$ 
```

#### ✅ Succeeds on a colleague's local platform
* `uname -a`
```
Linux localhost.localdomain 5.11.12-200.fc33.x86_64 #1 SMP Thu Apr 8 02:34:17 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
* `cat /etc/redhat-release`
```
Fedora release 33 (Thirty Three)
```
* Java 11
* docker-maven-plugin version: 0.30.0
* Docker version: engine 20.10.5, api 1.41
```
Client: Docker Engine - Community
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:19:16 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:16:56 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

```
Apache Maven 3.6.3 (Red Hat 3.6.3-5)
Maven home: /usr/share/maven
Java version: 11.0.10, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-11-openjdk-11.0.10.0.9-0.fc33.x86_64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.11.12-200.fc33.x86_64", arch: "amd64", family: "unix"
```

#### ✅ Succeeds on Travis
(see Travis [build log](https://travis-ci.org/github/OA-PASS/nihms-submission-etl/builds/766795585))

* Java 8
* docker-maven-plugin version: 0.30.0
* Docker version:
```
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:38 2017
 OS/Arch:      linux/amd64
Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:41:20 2017
 OS/Arch:      linux/amd64
 Experimental: false
```
* Maven version:
```
Apache Maven 3.5.2 (138edd61fd100ec658bfa2d307c43b76940a5d7d; 2017-10-18T07:58:13Z)
Maven home: /usr/local/maven-3.5.2
Java version: 1.8.0_151, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-98-generic", arch: "amd64", family: "unix"
```

* If it's a bug, how to reproduce :
* If it's a feature request, what is your use case :
* Sample project : *[GitHub Clone URL]*
